pytest_inmanta/plugin.py

```diff
@@ -247,14 +254,54 @@ def __init__(self, uri):
         self._env_id = cfg_env.get()
 
 
+class InmantaPluginsImportLoader:
+    """
+        Makes inmanta_plugins packages (Python source for inmanta modules) available dynamically so that tests can use them
+        safely without having to refresh imports when the compiler is reset.
+    """
+    def __init__(self, importer: "InmantaPluginsImporter") -> None:
+        self._importer: InmantaPluginsImporter = importer
+
+    def __getattr__(self, name: str):
+        submodules: Optional[Dict[str, ModuleType]] = self._importer.get_submodules(name)
+        fq_mod_name: str = f"inmanta_plugins.{name}"
+        if submodules is None or fq_mod_name not in submodules:
+            raise AttributeError("No inmanta module named %s" % name)
+        return submodules[fq_mod_name]
+
+
+class InmantaPluginsImporter:
+    def __init__(self, project: "Project") -> None:
+        self.project: Project = project
+        self.loader: InmantaPluginsImportLoader = InmantaPluginsImportLoader(self)
+
+    def get_submodules(self, module_name: str) -> Optional[Dict[str, ModuleType]]:
+        inmanta_project: module.Project = module.Project.get()
+        if not inmanta_project.loaded:
+            raise Exception(
+                "Dynamically importing from inmanta_plugins requires a loaded inmanta.module.Project. Make sure to use the"
+                " project fixture."
+            )
+        modules: Dict[str, module.Module] = inmanta_project.get_modules()
+        importlib.invalidate_caches()
+        loader.configure_module_finder(self.project.module_path)
```

Comment:
I don't think this line is required. When `project.load()` is invoked, the module_finder should be configured correctly. Is that correct?

Reply:
I agree, will fix.